verification/src/block_verifier.rs

```diff
@@ -447,26 +448,30 @@ impl<P: ChainProvider + CellProvider> TransactionsVerifier<P> {
         };
 
         let parent_hash = block.header().parent_hash();
+        let max_cycles = self.provider.consensus().max_block_cycles();
         // make verifiers orthogonal
         // skip first tx, assume the first is cellbase, other verifier will verify cellbase
-        let err: Vec<(usize, TransactionError)> = block
+        let results: Vec<Result<Cycle, TransactionError>> = block
             .commit_transactions()
             .par_iter()
             .skip(1)
             .map(|x| wrapper.resolve_transaction_at(x, &parent_hash))
+            .map(|tx| TransactionVerifier::new(&tx).verify(max_cycles))
+            .collect();
```

Comment:
Do we need to collect all the errors here? I think [while_some](https://docs.rs/rayon/1.0.3/rayon/iter/trait.ParallelIterator.html#method.while_some) can be used here to return early. However, this is not related to this PR.

For this PR, we also can use some advanced methods from rayon, such as [try_fold](https://docs.rs/rayon/1.0.3/rayon/iter/trait.ParallelIterator.html#method.try_fold) to return early when the total cycles exceed the limit. Also, see other methods starting with `try_` if `try_fold` does not work.

Reply:
@doitian @zhangsoledad How about now?